src/aspire/source/image.py

```diff
@@ -27,7 +29,50 @@
 logger = logging.getLogger(__name__)
 
 
-class ImageSource:
+class ImageAccessor:
+    """
+    Helper class for accessing images from an ImageSource as slices via the `src.images[start:stop:step]` API.
+    """
+
+    def __init__(self, fun, num_imgs):
+        """
+        :param fun: The private image-accessing method specific to the ImageSource associated with this ImageAccessor.
+                    Generally _images() but can be substituted with a custom method.
+        :param num_imgs: The max number of images that this ImageAccessor can load (generally ImageSource.n).
+        """
+        self.fun = fun
+        self.num_imgs = num_imgs
+
+    def __getitem__(self, indices):
+        """
+        ImageAccessor can be indexed via Python slice object, 1-D NumPy array, list, or a single integer,
+        corresponding to the indices of the requested images. By default, slices default to a start of 0,
+        an end of self.num_imgs, and a step of 1.
+        :return: An Image object containing the requested images.
+        """
+        if isinstance(indices, Iterable) and not isinstance(indices, np.ndarray):
+            indices = np.fromiter(indices, int)
+        elif isinstance(indices, (int, np.integer)):
+            indices = np.array([indices])
+        elif isinstance(indices, slice):
+            start, stop, step = indices.start, indices.stop, indices.step
+            if not start:
+                start = 0
+            if not stop or stop > self.num_imgs:
+                stop = self.num_imgs
+            if not step:
+                step = 1
+            indices = np.arange(start, stop, step)
+        if not isinstance(indices, np.ndarray):
+            raise KeyError(
+                "Key for .images must be a slice, 1-D NumPy array, or iterable yielding integers."
+            )
+        if not indices.ndim == 1:
+            raise KeyError("Only one-dimensional indexing is allowed for images.")
+        return self.fun(indices)
```

Comment:
How do we deal with negative indices (in slice, iterable, or vector form)? Does this need to be coded explicitly? If not, we should probably add tests to make sure that it works correctly.

Reply:
This is a potentially larger issue, as the downstream `_images` methods aren't currently set up to handle negative indices.. Looking into it

Reply:
Also added tests for negative indices and ranges